File: frontend/src/api/models/survey.ts
export interface QuestionType {
  id: number;
  name: string;
}

export interface AnswersSurvey {
  id: number;
  status: AnswerSurveyStatus;
  user_id: number;
  answered_questions?: Question[];
  current_question_index?: number;
  questions: Question[];
  not_answered_questions?: Question[];
}

export interface OptionQuestion {
  id: number;
  name: string;
  correct?: boolean;
}

export interface Question {
  id: number;
  name: string;
  question_type: QuestionType;
  options?: OptionQuestion[];
  image_url?: string;
}

export interface Survey {
  answers_surveys: AnswersSurvey[] | null;
  current_answers_survey: AnswersSurvey | null;
  id: number;
  name: string;
  description: string;
  questions: Question[];
  survey_subject_id: number;
  total_questions_quantity: number;
  image_url?: string;
  icon_url?: string;
}

export enum AnswerSurveyStatus {
  NotStarted = 'Not started',
  Started = 'Started',
  Completed = 'Completed',
}

File: android-client/Espoo/app/src/main/java/com/espoo/android/model/Survey.kt
package com.espoo.android.model

import com.google.gson.annotations.SerializedName

data class Survey(var id: Int, @SerializedName("survey_subject_id") var surveySubjectId: Int,
                  var name: String, var description: String, @SerializedName("role_type") var roleType: String,
                  @SerializedName("current_answers_survey") var currentAnswersSurvey: SimpleAnswersSurvey,
                  @SerializedName("answers_surveys") var answersSurvey: Array<SimpleAnswersSurvey>, var questions: Array<Question>)

data class SimpleSurvey(var id: Int, var name: String, var description: String, @SerializedName("questions_quantity") var questionsQuantity: Int)

data class SimpleAnswersSurvey(var id: Int, @SerializedName("user_id") var userId: Int, var status: String)

data class Question(var id: Int, var name: String, var options: Array<Option>)

data class Option(var id: Int, var name: String, var correct: Boolean)